config/config.go

```diff
@@ -113,17 +112,46 @@ func defaultConfigData() *configurationData {
 
 // NewConfig returns a configuration
 func NewConfig() (*Config, error) {
-	filename := os.Getenv("SNOWBRIDGE_CONFIG_FILE")
-	if filename == "" {
-		return newEnvConfig()
-	}
 
-	switch suffix := strings.ToLower(filepath.Ext(filename)); suffix {
-	case ".hcl":
-		return newHclConfig(filename)
+	var config *Config
+	var err error
+
+	// If we don't have a config file path, use env var configuration.
+	// Otherwise, use the file.
+	switch filename := os.Getenv("SNOWBRIDGE_CONFIG_FILE"); filename {
+	case "":
+		config, err = newEnvConfig()
+		if err != nil {
+			return nil, errors.Wrap(err, "Error parsing env config")
+		}
+
+		// If we have set TRANSFORM_CONFIG_B64, parse it into a config and overwrite the Transformation portion of our existing config.
+		if b64Transformations := os.Getenv("TRANSFORM_CONFIG_B64"); b64Transformations != "" {
+			transformationEnvConfig, err := base64.StdEncoding.DecodeString(b64Transformations)
+			if err != nil {
+				return nil, errors.Wrap(err, "Error decoding b64 data from TRANSFORM_CONFIG_B64 env var")
+			}
+
+			transformationConfig, err := newHclConfig(transformationEnvConfig, "")
+			if err != nil {
+				return nil, errors.Wrap(err, "Error creating config fom provided TRANSFORM_CONFIG_B64")
+			}
+
+			config.Data.Transformations = transformationConfig.Data.Transformations
+		}
+
+		return config, err
+
 	default:
-		return nil, errors.New("invalid extension for the configuration file")
```

Comment:
Assuming you are no longer worried about the file extension for the config and its just assumed to be HCL and it will fail if its not?

Reply:
Yeah exactly - there was never any reason to check the extension - as long as it contains valid hcl we can build the config. So I just removed the check.

Your asking did make me realise that this is a behaviour change though, and so I've added a test case to cover this behaviour. :)